packages/react-instantsearch/src/connectors/connectScrollTo.js

```diff
@@ -32,6 +38,36 @@ export default createConnector({
       null,
       currentRefinement => currentRefinement
     );
-    return { value };
+
+    if (!this._prevSearchState) {
+      this._prevSearchState = {};
+    }
+
+    /* Get the subpart of the state that interest us*/
+    if (hasMultipleIndex(this.context)) {
+      const index = getIndex(this.context);
+      searchState = searchState.indices ? searchState.indices[index] : {};
+    }
+
+    /*
+      We need to keep track of the search state to know if there's a change in the app
+      that was not triggered by the props.scrollOn (id) or the Configure widget. 
```

Comment:
Maybe "if there is a change in the app that has been triggered by another element than "props.scrollOn (id) or the Configure widget" to insist on the fact that only those 2 are able to affect scrollTo